backend/bitboxbase/bitboxbase.go

```diff
@@ -44,17 +48,49 @@ type Interface interface {
 	GetRegisterTime() time.Time
 
 	// MiddlewareInfo returns some blockchain information.
-	MiddlewareInfo() (rpcclient.SampleInfoResponse, error)
+	MiddlewareInfo() (rpcmessages.SampleInfoResponse, error)
+
+	// VerificationProgress returns the bitcoind verification progress.
+	VerificationProgress() (rpcmessages.VerificationProgressResponse, error)
 
 	// ConnectElectrum connects to the electrs server on the base and configures the backend accordingly
 	ConnectElectrum() error
 
 	// Ping sends a get requset to the bitbox base middleware root handler and returns true if successful
 	Ping() (bool, error)
+
+	// Self returns an instance of the base
+	Self() *BitBoxBase
+
+	// Status returns the current status of the base
+	Status() bitboxbasestatus.Status
+
+	// ChannelHash returns the hash of the noise channel
+	ChannelHash() (string, bool)
+
+	// Deregister calls the backend's BitBoxBase Deregister callback and sends a notification to the frontend, if bitboxbase is active.
+	// If bitboxbase is not active, an error is returned.
+	Deregister() (bool, error)
+
+	// SyncOption returns true if the bitcoin resync rpc call executed successfully
+	SyncOptionStatus(SyncOption) (bool, error)
 }
 
+// SyncOption is a user provided blockchain sync option during BBB initialization
+type SyncOption int
+
+// SyncOption iota has three options:
+// Accept pre-synchronized blockchain; delete the chainstate and reindex; resync bitcon from scratch with an IBD
+const (
+	PRESYNCED SyncOption = iota
```

Comment:
Call them `SyncOptionPresynced`, `SyncOptionReindex`, `SyncOptionInitianBlockDownload`

Need `SyncOption` prefix for namespacing.

Also pls put the documentation for each var right before the var in godoc style, like in `event.go`

Reply:
ack, changed from iota to strings